internal/datanode/compactor.go

```diff
@@ -211,6 +213,14 @@ func (t *compactionTask) merge(mergeItr iterator, delta map[UniqueID]Timestamp,
 			continue
 		}
 
+		if Params.DataCoordCfg.EnableAutoExpiration {
```

Comment:
maybe add some stats here? like how many entities are skipped during expiration

Reply:
I added logging for the number of expired entity on 9e003150d0b792e8bfafb73784aab4f3028d2354 :)